martian/src/metadata.rs

```diff
@@ -189,32 +188,19 @@ impl Metadata {
     /// The file write is fully atomic.
     pub(crate) fn write_metadata(&mut self, name: &str, content: &str) -> Result<()> {
         fully_atomic_write(&self.make_path(name), |w| w.write_all(content.as_bytes()))?;
-        self.update_journal(name)?;
-        Ok(())
+        self.update_journal(name)
     }
 
     /// Update the Martian journal -- so that Martian knows what we've updated
     fn update_journal(&self, name: &str) -> Result<()> {
-        let journal_name: Cow<'_, str> = if self.stage_type != StageType::Main {
-            format!("{}_{name}", self.stage_type).into()
+        let journal_filename = if self.stage_type == StageType::Main {
+            format!("{}.{name}", self.run_file)
         } else {
-            name.into()
+            format!("{}.{}_{name}", self.run_file, self.stage_type)
         };
-        let journal_file = format!("{}.{journal_name}", self.run_file);
-        let timestamp = make_timestamp_now();
-        let mut write_err = Ok(());
-        let create_result = fully_atomic_write(Path::new(&journal_file), |w| {
-            write_err = w.write_all(timestamp.as_bytes());
-            Ok(())
-        });
-        if let Err(err) = write_err {
-            // Pretty much ignore this error.  The only reason we need
-            // any content at all in this file is because some
-            // filesystems behave strangely with completely empty files.
-            eprintln!("Writing journal file {journal_file}.tmp: {err}");
-        }
-        create_result?;
-        Ok(())
+        fully_atomic_write(Path::new(&journal_filename), |w| {
+            w.write_all(make_timestamp_now().as_bytes())
+        })
```

Comment:
If we're going to do this, we should refactor `fully_atomic_write` - the only reason this was done with a closure was to enable this conditional ignoring of the timestamp write error for journal files.

Reply:
Done. `fully_atomic_write` now accepts a `str` rather than a closure.